api/loader/fileloader.go

```diff
@@ -127,20 +134,21 @@ func newLoaderOrDie(
 		log.Fatalf("unable to make loader at '%s'; %v", path, err)
 	}
 	return newLoaderAtConfirmedDir(
-		lr, root, fSys, nil, git.ClonerUsingGitExec)
+		lr, root, fSys, nil, git.ClonerUsingGitExec, getResource)
 }
 
 // newLoaderAtConfirmedDir returns a new fileLoader with given root.
 func newLoaderAtConfirmedDir(
 	lr LoadRestrictorFunc,
 	root filesys.ConfirmedDir, fSys filesys.FileSystem,
-	referrer *fileLoader, cloner git.Cloner) *fileLoader {
+	referrer *fileLoader, cloner git.Cloner, getter resourceGetter) *fileLoader {
```

Comment:
I looks like `getter` and cover the functionality of `cloner`. Can you see if we can use getter only rather than both?

Reply:
There is an intersection between getter and cloner. Some special URLs can only be handled by cloner at the moment.

One thing we can do is refactoring cloner to Detect and Get in getter and register them in getter. Then we shall have the unified interface while keeping backward compatibility.

What do you think?

Reply:
What are the special URLs that can't be handled by cloner? If they are not supported by getter, keeping the cloner for them should be fine.

Reply:
For example,
https://github.com/kubernetes-sigs/kustomize/blob/064f0641ba8c0b589a7df7c029031da78a6fdf47/api/internal/git/repospec.go#L51

I don't see detectors in go-getter to handle them. It supports only plain git, GitHub and bitbucket url.